File: 2021/day1/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func part1() {
	// read file
	bytes, err := os.ReadFile("day1/input.txt")
	if err != nil {
		panic(err)
	}
	data := strings.Split(string(bytes), "\n")

	count := 0
	prev := 999999
	for idx, number := range data {
		if number == "" {
			break
		}
		if idx == 0 {
			prev, err = strconv.Atoi(number)
			if err != nil {
				panic(err)
			}
			continue
		}
		n, err := strconv.Atoi(number)
		if err != nil {
			fmt.Println(idx, number)
			panic(err)
		}
		if n > prev {
			count++
		}

		prev = n
	}

	fmt.Printf("Part 1: %d\n", count)

}

func part2() {
	// read file
	bytes, err := os.ReadFile("day1/input.txt")
	if err != nil {
		panic(err)
	}
	data := strings.Split(string(bytes), "\n")

	count := 0
	prev1 := 9999999
	prev2 := 9999999
	prev3 := 9999999
	for idx, n := range data {
		if n == "" {
			break
		}
		number, err := strconv.Atoi(n)
		if err != nil {
			panic(err)
		}
		switch idx {
		case 0:
			prev3 = number
			continue
		case 1:
			prev2 = number
			continue
		case 2:
			prev1 = number
			continue
		}

		if (number + prev1 + prev2) > (prev1 + prev2 + prev3) {
			count++
		}
		prev1, prev2, prev3 = number, prev1, prev2
		fmt.Println(number, prev1, prev2, prev3)

	}

	fmt.Printf("Part 2: %d\n", count)

}

func main() {
	part1()
	part2()
}

File: 2021/day1/main.ts
import * as fs from "fs";


function part1(): void {

    // read file
    const data: string = fs.readFileSync("day1/input.txt", "utf-8")
    const lines: string[] = data.split("\n")

    let count: number = 0
    let prev: number = 999999

    for (let idx in lines) {
        const number = parseInt(lines[idx])
        if (parseInt(idx) == 0) {
            prev = number
            continue
        }
        if (isNaN(number)) {
            break
        }
        if (number > prev) {
            count++
        }

        prev = number

    }

    console.log("Part1: %d", count)

}

function part2(): void {
    // read file
    const data: string = fs.readFileSync("day1/input.txt", "utf-8")
    const lines: string[] = data.split("\n")

    let count: number = 0
    let prev1: number = 999999
    let prev2: number = 999999
    let prev3: number = 999999

    for (let idx in lines) {
        let number: number = parseInt(lines[parseInt(idx)])
        if (isNaN(number)) {
            break
        }
        switch (parseInt(idx)) {
            case 0:
                prev3 = number
                continue
            case 1:
                prev2 = number
                continue
            case 2:
                prev1 = number
                continue
        }

        if ((number + prev1 + prev2) > (prev1 + prev2 + prev3)) {
            count++
        }

        prev3 = prev2
        prev2 = prev1
        prev1 = number

    }

    console.log("Part2: %d", count)

}

function main(): void {
    part1()
    part2()
}

main()
